### PR TITLE
Remove TODO(3.0) tags in pkey ctrl functions.

### PR DESCRIPTION
**Pull request by slontis**
Added a legacy related macro for the pkey controls.
Moved dsa & rsa control functions out of XXX_lib.c into XXX_ctrl.c. These are excluded from the fips module.
Removed the auto increment past the end of param arrays in multiple places  (i.e *p = OSSL_PARAM_construct_end())

##### Checklist

- [ ] documentation is added or updated
- [ ] tests are added or updated

### CODE REVIEW

**Comment by slontis** [comment]
ping

File: crypto/rsa/rsa_ctrl.c [hunk 1]
```diff
@@ -0,0 +1,598 @@
...
+
+    *p++ = OSSL_PARAM_construct_utf8_string(OSSL_ASYM_CIPHER_PARAM_OAEP_DIGEST,
+                                            /*
+                                             * Cast away the const. This is read
+                                             * only so should be safe
+                                             */
+                                            (char *)mdname, 0);
+    if (mdprops != NULL) {
+        *p++ = OSSL_PARAM_construct_utf8_string(
+                    OSSL_ASYM_CIPHER_PARAM_OAEP_DIGEST_PROPS,
```
Comment (levitte) [inline crypto/rsa/rsa_ctrl.c:136]: Not sure about this source style

Comment (slontis) [reply crypto/rsa/rsa_ctrl.c:136]: assigned to a variable - not much else can be done when everything is so verbose..

**Comment by slontis** [comment]
ping

**Comment by slontis** [comment]
ping

**Comment by t8m** [comment]
@slontis Shane, I'm taking this PR over from you. Are you OK with it?

**Comment by slontis** [comment]
Yes that is fine..
I am not sure if you agree with what I was doing in here..
Will be interesting to see what you do. Should not take long to fix this one up hopefully.

**Comment by levitte** [comment]
Rebase needed, plus this competes with #13913.  If you can do a fairly quick job of this, @t8m, I have no problems adapting my work to whatever comes out.

**Comment by t8m** [comment]
Hmm, looking at the #13913 makes me think that this PR should be simply closed as the approach taken there is basically reverse of what is done here. There are some useful nits among the changes here that could be applied after #13913 from this PR but otherwise it does not make much sense. So I'll probably postpone this after the #13913 is merged.

**Comment by levitte** [comment]
> ...  So I'll probably postpone this after the #13913 is merged.

Okie, makes sense.

**Comment by levitte** [comment]
#13913 is merged.  Time to rebase this and see what's left of it, if any...

**Comment by t8m** [comment]
> #13913 is merged. Time to rebase this and see what's left of it, if any...

I am just going through this and applying remaining useful changes from this to a new PR. Then I'll close this one.

**Comment by t8m** [comment]
This is now superseded by #14290

**Pull request closed without merging**